xcell/mappers/mapper_base.py

```diff
@@ -35,3 +36,12 @@ def get_nmt_field(self, **kwargs):
             self.nmt_field = nmt.NmtField(mask, signal,
                                           templates=cont, n_iter=n_iter)
         return self.nmt_field
+
+    def _get_nmt_field(self, signal=None, **kwargs):
```

Comment:
Just a reminder for myself to figure out why you need this new method. No need to do anything.

Reply:
OK, I understand now. Instead of creating a new method, why don't we modify `get_nmt_field` as:
```
if 'signal' in kwargs:
    signal = kwargs['signal']
else:
    signal = self.get_signal_map(**kwargs)
```
and you could do the same with the contaminants.

Reply:
it is also that get_mnt_field() updates the class object mnt_field and here we don't want that. 
Do you think we should still unify the methods?

Reply:
OK, so we put all the code in `_get_nmt_field` and then `get_nmt_field` is just a wrapper of it